### DISCUSSION TITLE
How to read/set field setting values?

### DISCUSSION THREAD

**Discussion by ywteh** [omerfaruk-aran/esphome_samsung_hvac_bus#189] [Questions]
Hi,

I managed to follow instructions and got ESPHOME set up to talk to my Samsung heat pump (R290). I'm trying to control some of the field setting values (FSVs) using the system, by checking up the information at https://wiki.myehs.eu/wiki/NASA_Protocol. Specifically, I'd like to control the water law high/low temperatures and the water outlet target temperature, but to no avail. 

I have forked the code here: https://github.com/ywteh/esphome_samsung_ac . The relevant code can be found if you search for "water_law_hi_temp", "water_law_lo_temp" and "water_outlet_target". 

The entities do appear on home assistant, but their values are "unknown" and I cannot adjust their values using home assistant. I'm wondering if it is possible to use esphome_samsung_ac to set FSVs, or if I did something wrong?

Thank you very much, your help is much appreciated.

cheers,
-yw

**Comment by brazoayeye** [comment]
Hello, have you seen the packet number in the network using the debug log?

I made a fork time ago to add a custom climate, and for me (R32) it's working without problems. 

Have a look in my configuration file for addresses and double check you inserted the right device as described also in this repo. If you log messages you can post the results

**Reply by Beormund** [reply]
Do you need to issue a command to read FSVs? I'm not sure these are published unless you request them to be.

**Reply by brazoayeye** [reply]
In my heat pump all sent messages are unsolicited. EspHome device only read incoming messages and write data to devices registers. I read that with non nasa device something is/could be different, or maybe there is something i'm missing.

**Reply by Beormund** [reply]
I created custom sensors for messages 0x4254 - 0x4257 which are to do with Water Law (weather compensation). But nothing is received.

**Reply by Beormund** [reply]
@brazoayeye, @lanwin, @ywteh   The wifi receiver appears to issue a request to read values. I noticed this in the logs:

| Time | Level | Tag | Message |
|---|---|---|---|
| 11:47:31 | [W] | [samsung_ac:877] | Read #Packet Src:62.00.6e Dst:b2.00.20 {PacketInformation: 1;ProtocolVersion: 2;RetryCount: 0;PacketType: 1;DataType: 1;PacketNumber: 248}  <br>> Enum 4093 = 0  <br>> Enum 4094 = 0  <br>> Variable 4273 = 0  <br>> Variable 4274 = 0  <br>> Variable 4275 = 0  <br>> Variable 4276 = 0  <br>> Variable 4277 = 0  <br>> Variable 4278 = 0  <br>> Variable 4279 = 0  <br>> Variable 427a = 0  <br>> Variable 427b = 0 |
| 11:47:31 | [W] | [samsung_ac:862] | Response #Packet Src:20.00.00 Dst:62.00.6e {PacketInformation: 1;ProtocolVersion: 2;RetryCount: 0;PacketType: 1;DataType: 5;PacketNumber: 248}  <br>> Enum 4093 = 1  <br>> Enum 4094 = 1  <br>> Variable 4273 = 250  <br>> Variable 4274 = 300  <br>> Variable 4275 = 150  <br>> Variable 4276 = 160  <br>> Variable 4277 = 250  <br>> Variable 4278 = 250  <br>> Variable 4279 = 150  <br>> Variable 427a = 150  <br>> Variable 427b = 300 |

It was requesting Outing FSV values. I'd like to be able to do the same for Water Law FSVs.

**Reply by Beormund** [reply]
@lanwin @brazoayeye  What would be great would be to have something like this via YAML:

```yaml
- request_read
   update_interval: 60s
   messages:
     - 0x4253
     - 0x4254
```

**Reply by mergwyn** [reply]
I would also be interested in getting access to some of the FSV values (particularly around Water Law).  I've turned debguuung undefined messages on and don;t see them n the logs. Does that mean that they are not available in the nasa protocol for my device, or that something specific such as suggested by  @Beormund needs to be done to request additional values?

**Reply by brazoayeye** [reply]
If you want to reproduce something that's feasible from local device do that while logging and check moving messages. 

Then idk if you have already tried what described on https://github.com/omerfaruk-aran/esphome_samsung_hvac_bus/discussions/82#discussioncomment-11907883
